patches/minecraft/net/minecraft/entity/EntityClassification.java.patch

```diff
@@ -9,13 +9,20 @@
     MONSTER("monster", 70, false, false, 128),
     CREATURE("creature", 10, true, true, 128),
     AMBIENT("ambient", 15, true, false, 128),
-@@ -57,6 +57,10 @@
+@@ -57,6 +57,17 @@
        return this.field_82707_i;
     }
  
 +   public static EntityClassification create(String name, String id, int maxNumberOfCreatureIn, boolean isPeacefulCreatureIn, boolean isAnimalIn, int despawnDistance) {
 +      throw new IllegalStateException("Enum not extended");
 +   }
++
++   @Override
++   @Deprecated
++   public void init() {
++      field_220364_f.put(this.field_220365_j, this);
++      field_233667_g_ = IStringSerializable.func_233023_a_(EntityClassification::values, EntityClassification::func_233670_a_);
```

Comment:
I have my doubts if this second line is actually needed as the Codec seems to reference `Enum#values` and not the cached `VALUES_MAP`.

Reply:
yes but it caches it instantly.

Reply:
the latter would probably be the better solution as it would also resolve this for other extensible enums that use this (currently only `net.minecraft.world.gen.feature.jigsaw.JigsawPattern.PlacementBehaviour` though).
But it is more "intrusive", although I can't see any issues on first sight/test), I am not going to make that call

Reply:
Makeing the codec a public non-final is a non-starter.
Perhaps the better solution would be to introduce a new codec type for extensible enums?
And then change the codec to use that instead of the default enum one.
Or, we could change the default enum one {if its in MC code} to not cache the values.
Either option is better then letting all modders change the entire codec object willy nilly.

Reply:
ok, removed the caching as Cybogmas suggested, but just for the two enums that are `IExtensibleEnum`